src/services/completions.ts

```diff
@@ -4062,10 +4062,14 @@ function getCompletionData(
         return scope;
     }
 
+    function isInDifferentLineWithContextToken(contextToken: Node, position: number): boolean {
+        return sourceFile.getLineEndOfPosition(contextToken.getEnd()) < position;
+    }
     function isCompletionListBlocker(contextToken: Node): boolean {
         const start = timestamp();
         const result = isInStringOrRegularExpressionOrTemplateLiteral(contextToken) ||
-            isSolelyIdentifierDefinitionLocation(contextToken) ||
+            // GH#54729 ignore this case when current position is in a newline

```

Comment:
I think, if we wanted to be more precise in when we allow completions, this check of `isInDifferentLineWithContextToken` should be done inside `isSolelyIdentifierDefinitionLocation`, because I don't think we want to allow completions in a newline for every possible case `isSolelyIdentifierDefinitionLocation` returns true.

For instance, `isSolelyIdentifierDefinitionLocation` returns true on the following position:
```ts
var [x, y
|
```
It doesn't look like you'd actually want completions at that position.

Reply:
Thank you! Tried to fix this, @gabritto plz take a look at the new code :)